prov/efa/src/rxr/rxr_pkt_cmd.c

```diff
@@ -671,9 +671,155 @@ void rxr_pkt_handle_data_copied(struct rxr_ep *ep,
 	}
 }
 
-/*
- *   Functions used to handle packet send completion
+/**
+ * @brief handle the a packet that encountered error completion while sending
+ *
+ * If the error is RNR, this function will queue a packet if necessary. It
+ * determines whether to queue a RNR packet using the following rule:
+ *
+ *	In general, RNR packet will be queued and resend by the progress enginee.
+ *
+ *	It is NOT queued only if application want to receive an RNR error CQ entry
+ *	for an send operation.
+ *
+ * For other types of error, the function will write cq/eq error entry then
```

Comment:
you forgot to mention that there are packets that will never return to application here.

Reply:
Sorry. I am not sure I understand. You mean for some packet, the device will not return completion to EFA provider? or EFA provider will not return completion to application?

Reply:
I liked your description in the commit message better than this comment in terms of figuring out when the events occurred.

Reply:
Thanks for clarifying! I update the comments by rewording the commit message.